department-of-veterans-affairs/abd-vro
# MCP-1983: Incorporate Mas processing into Camel routes and post status

## What was the problem?

1. The code for accessing MAS, collecting the results, and generating PDF is in one class rather than spread out into uncoupled camel routes
2. Auditing is done at the camel endpoint level, so with the code in one piece it was hard to audit the separate substeps
3. If the MAS API collection is not ready, we keep retrying it indefinitely. We need to only try a fixed number of times

Associated tickets or Slack threads:

- [MCP-1922](https://amida.atlassian.net/browse/MCP-1922)
- [MCP-1933](https://amida.atlassian.net/browse/MCP-1933)
- [MCP-2015](https://amida.atlassian.net/browse/MCP-2015)
- [MCP-2016](https://amida.atlassian.net/browse/MCP-2016)

## How does this fix it?[^1]

- Refactored the code into individual camel routes and added auditing
- Introduced the configurable parameter mas-processing-retry-count. We will only check if the collection is processed this number of times and then exit.

## How to test this PR
- Invoke the /v1/automatedClaim route with a valid request that MAS can process
- Examine the logs to make sure the request gets processed
- Examine the database to see the various events logged in the audit_event table

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
